rust/crd/src/lib.rs

```diff
@@ -98,54 +104,97 @@ pub struct SparkApplicationStatus {
 )]
 #[serde(rename_all = "camelCase")]
 pub struct SparkApplicationSpec {
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub version: Option<String>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub mode: Option<String>,
+    /// Mode: cluster or client. Currently only cluster is supported.
+    pub mode: String,
```

Comment:
Nit: While we already have a breaking change we could switch to an enum directly ;)

Reply:
See 2a1bb77.